From: Andrew Morton <akpm@linux-foundation.org>
Subject: arch/x86/include/asm/pgtable.h needs spinlock.h

In file included from ./include/linux/pgtable.h:6,
                 from ./include/linux/kasan.h:30,
                 from mm/kasan/common.c:14:
./arch/x86/include/asm/pgtable.h:56:8: error: unknown type name 'spinlock_t'
   56 | extern spinlock_t pgd_lock;
      |        ^~~~~~~~~~


Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/x86/include/asm/pgtable.h |    1 +
 mm/kasan/common.c              |    1 +
 2 files changed, 2 insertions(+)

--- a/arch/x86/include/asm/pgtable.h~arch-x86-include-asm-pgtableh-needs-spinlockh
+++ a/arch/x86/include/asm/pgtable.h
@@ -3,6 +3,7 @@
 #define _ASM_X86_PGTABLE_H
 
 #include <linux/mem_encrypt.h>
+#include <linux/spinlock.h>
 #include <asm/page.h>
 #include <asm/pgtable_types.h>
 
--- a/mm/kasan/common.c~arch-x86-include-asm-pgtableh-needs-spinlockh
+++ a/mm/kasan/common.c
@@ -15,6 +15,7 @@
 #include <linux/kernel.h>
 #include <linux/linkage.h>
 #include <linux/memblock.h>
+#include <linux/spinlock.h>
 #include <linux/memory.h>
 #include <linux/mm.h>
 #include <linux/module.h>
_
